compiler/sigmatch.nim

```diff
@@ -2371,6 +2371,10 @@ proc matchesAux(c: PContext, n, nOrig: PNode, m: var TCandidate, marker: var Int
       elif not isLValue(c, n):
         m.firstMismatch.kind = kVarNeeded
         noMatch()
+    if formal.typ.kind == tyGenericParam and formal.typ.len > 0 and formal.typ[0].kind == tyOr:
+      for s in formal.typ[0].sons:
+        if compareTypes(n.typ, s):
+          inc(m.genericMatches, 1)
```

Comment:
Correct me if I'm wrong but this solution only happens to work because `SomeInteger` contains more types than `SomeFloat`.

Reply:
in the test case will `compareTypes` with `int` first, the edge case will be `SomeFloat` contains more than one `int`

Reply:
furthermore here can make `alt` not match , not just make score higher

Reply:
after debugging `compareTypes` not cover all case, so result to noMatch is not reliable. score comparision is fine for now.